src/beacon.cpp

```diff
@@ -225,3 +201,38 @@ bool VerifyBeaconContractTx(const CTransaction& tx)
     // Passed checks
     return true;
 }
+
+bool ImportBeaconKeysFromConfig()
+{
+    AssertLockHeld(cs_main);
+    std::string sBeaconPublicKey = GetBeaconPublicKey(GlobalCPUMiningCPID.cpid,false);
+    std::string sCPID(msPrimaryCPID);
+    std::string strSecret= GetArgument("privatekey" + sCPID + (fTestNet ? "testnet" : ""), "");
+    if(strSecret.empty())
+        return false;
+    auto vecsecret = ParseHex(strSecret);
+
+    CKey key;
+    if(!key.SetPrivKey(CPrivKey(vecsecret.begin(),vecsecret.end())))
+        return error("ImportBeaconKeysFromConfig: Invalid private key");
```

Comment:
Hmm, the caller won't be able to distinguish between success (true) or an error (not null -> true).

Reply:
Don't review when tired. `error` returns false.